specification/transport/abstract.tex

```diff
@@ -693,13 +696,30 @@ \subsubsection{Behaviors}
 any other successfully reassembled transfer under the same session specifier by more than the transfer-ID timeout
 is considered unique regardless of its transfer-ID value.
 
+If the optimal transfer-ID timeout value for a given session cannot be known in advance,
+it can be computed at runtime as a multiple of the average transfer reception interval (on this session).
+The parameters of such computation are to be chosen according to the requirements of the application,
+but they should always be documented.
+
 \begin{remark}
     Low transfer-ID timeout values increase the risk of undetected transfer duplication when such transfers
     are significantly delayed due to network congestion,
     which is possible with very low-priority transfers when the network load is high.
 
     High transfer-ID timeout values increase the risk of an undetected transfer loss
     when a remote node suffers a loss of state (e.g., due to a software reset).
+
+    The ability to auto-detect the optimal transfer-ID timeout value per session at runtime ensures that the
+    application can find the optimal balance even if the temporal properties of the network are not known in advance.
+    As a practical example, an implementation could compute the exponential moving average of the
+    transfer reception interval $x$ for a given session and define the transfer-ID timeout as $2x$.
+
+    It is important to note that the automatic adjustment of the transfer-ID timeout should only be done
+    on a per-session basis rather than for the entire port, because there may be multiple remote nodes
```

Comment:
To use a per-session average you need to start with a non-zero number so there's still some hard-coded default involved, correct?

Reply:
I feel like we're also poking around at network congestion algorithms here. Is this something where I could implement a slow-start algorithm on the transmitter such that the receiver would get a slower but more consistent pacing of frames for a given transfer?

Reply:
> To use a per-session average you need to start with a non-zero number so there's still some hard-coded default involved, correct?

Yes, but since we allow the application to override the transfer-ID, by extension we also allow it to choose an arbitrary initial value. That's how I intended it to read anyway.

> I feel like we're also poking around at network congestion algorithms here. Is this something where I could implement a slow-start algorithm on the transmitter such that the receiver would get a slower but more consistent pacing of frames for a given transfer?

You seem to be talking about traffic shaping (leaky bucket) logic here, but I don't think it is related to transfer-ID timeout. I expect the importance of traffic policy enforcement to decrease in a network that segregates traffic by priority levels (which is why it is rarely talked about in the context of CAN networks where there are 2^(29 or 11) priority levels). The reason being that if a high-priority class traffic unconditionally preempts low priority traffic, the latter can be excluded from analysis at this level. I expect that we will speak about network calculus at length when we get to switched networks, that is a separate topic. CAN is much simpler in this regard.